File: 0500-keyboard-row/0500-keyboard-row.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var findWords = function(words) {
    const r1 = new Set("qwertyuiop");
    const r2 = new Set("asdfghjkl");
    const r3 = new Set("zxcvbnm");

    let result = [];

    for (let word of words) {
        let lower = word.toLowerCase();

        let row;
        if (r1.has(lower[0])) row = r1;
        else if (r2.has(lower[0])) row = r2;
        else row = r3;

        let valid = true;
        for (let ch of lower) {
            if (!row.has(ch)) {
                valid = false;
                break;
            }
        }

        if (valid) result.push(word);
    }

    return result;
};

File: 0500-keyboard-row/0500-keyboard-row.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        r1 = set("qwertyuiop")
        r2 = set("asdfghjkl")
        r3 = set("zxcvbnm")

        res = []
        
        for word in words:
            lower_word = word.lower()
            word_set = set(lower_word)

            if word_set.issubset(r1) or word_set.issubset(r2) or word_set.issubset(r3):
                res.append(word)

        return res
            